library/kani/src/lib.rs

```diff
@@ -140,6 +140,25 @@ pub const fn assert(cond: bool, msg: &'static str) {
 #[rustc_diagnostic_item = "KaniCover"]
 pub const fn cover(_cond: bool, _msg: &'static str) {}
 
+/// This function is only used for loop contract annotation.
+/// It behaves as a placeholder to telling us where is the loop head and loop end
+/// that we want to annotate to.
+#[inline(never)]
+#[rustc_diagnostic_item = "KaniLoopInvariant"]
+pub const fn kani_loop_invariant(_cond: bool) {}
```

Comment:
These functions seemed to be used as markers, so maybe rename that so it is explicit. This one for example seems to be marking the end of the loop, so it could be something like.
```rust
pub const fn kani_loop_end_marker() {}
```